middleware/auth.js

```diff
@@ -4,54 +4,44 @@ const ValidationUtil = require('../util/validationUtil');
 
 const HttpStatus = require('http-status-codes');
 
-const Auth = (req, res, next) => {
-    const token = req.headers['x-access-token'] || req.headers['authorization'];
 
-    // Decode token
-    if (token) {
-        // Remove unwanted prefix if exists
-        if (token.startsWith('Bearer ')) {
-            token = token.slice(7, token.length);
-        }
-        // Verify token
-        jwt.verify(token, AppConfig.SECRET, function (err, decoded) {
-            if (err) {
-                return res.status(HttpStatus.UNAUTHORIZED).json({ success: false, message: 'Unauthorized access.' });
-            } else {
-                req.decoded = decoded;
-                next();
-            }
-        });
-    } else {
-        return res.status(HttpStatus.FORBIDDEN).json({ success: false, message: 'No token provided.' });
-    }
-}
+const auth = (checkAccessType) => {
+    return (req, res, next) => {
```

Comment:
How are the `req`, `res` values and the `next` callback passed to the `auth` function?